source/common/http/filter_manager.cc

```diff
@@ -466,19 +466,35 @@ void ActiveStreamDecoderFilter::encode1xxHeaders(ResponseHeaderMapPtr&& headers)
   }
 }
 
+void ActiveStreamDecoderFilter::maybeMarkDecoderFilterTerminal(bool encoded_end_stream) {
```

Comment:
Changed code so that only terminal filter can work with independent half-close. One runtime key is removed the FM will agnostic to the half-close behavior - it will always assume independent half-close. And semantics of closing incomplete streams will move into codecs.